docs/switching-between-main-and-replica.md

```diff
@@ -0,0 +1,19 @@
+## How DualConnection chooses between main and replica?
+
+[DualConnection](../src/main/java/com/atlassian/db/replica/api/DualConnection.java) takes multiple aspects while deciding
+which connection to use:
+
+1. A [connection's state](dual-connection-states.md).
+2. A Replica's [consistency](../src/main/java/com/atlassian/db/replica/spi/ReplicaConsistency.java).
+3. Context of `java.sql.Connection`/`java.sql.Statement` API usage.
+
+Some of the methods are intended to write into the database. For example every call to `java.sq.PreparedStatement#executeUpdate`
+will switch the connection's state to the main database.
+
+4. Availability of replica
+5. The query will use the main database in case it's:
```

Comment:
Should we mention something about nested/complex queries?

Reply:
What do you mean?

Reply:
This mentions simple cases only. I remember we are using "startsWith" to check if there is DELETE/UPDATE etc.
Can there be a query that starts with select, but it's doing update later on? e.g. https://stackoverflow.com/a/6764367

Reply:
Yes. Not all the cases are explained here. The tests are more detailed documentation, and in case you find a nice corner case I encourage you to contribute a test :). See https://github.com/atlassian-labs/db-replica/blob/19b83af9838413c414cb64c238b7ff819082394b/src/test/java/com/atlassian/db/replica/api/TestDualConnection.java#L235, for example.